src/frontend/src/lib/utils/tokens.utils.ts

```diff
@@ -40,3 +44,60 @@ export const sortTokens = ({
 			a.name.localeCompare(b.name) ||
 			a.network.name.localeCompare(b.network.name)
 	);
+
+/**
+ * Pins tokens by USD value, balance and name.
+ *
+ * The function pins on top of the list the tokens that have a balance and/or an exchange rate.
+ * It sorts first the ones that have an exchange rate and balance non-zero, according to their usd value (descending).
+ * Then, it sorts the ones that have only a balance non-zero and no exchange rate, according to their balance (descending).
+ * Finally, it leaves the rest of the tokens untouched.
+ * In case of a tie, it sorts by token name and network name.
+ *
+ * @param $tokens - The list of tokens to sort.
+ * @param $exchanges - The exchange rates for the tokens.
+ * @param $balancesStore - The balances for the tokens.
+ * @returns The sorted list of tokens.
+ *
+ */
+export const pinTokensWithBalanceAtTop = ({
+	$tokens,
+	$balancesStore
+}: {
+	$tokens: TokenUi[];
+	$balancesStore: CertifiedStoreData<BalancesData>;
+}): TokenUi[] => {
+	const tokensWithBalanceToPin: TokenToPin[] = $tokens
+		.reduce(
+			(acc, token) => {
+				const balance = Number(
+					formatToken({
+						value: $balancesStore?.[token.id]?.data ?? BigNumber.from(0),
+						unitName: token.decimals,
+						displayDecimals: token.decimals
+					})
+				);
+
+				const usdBalance = token.usdBalance ?? 0;
+
+				if (usdBalance > 0 || balance > 0) {
+					acc.push({ ...token, balance });
```

Comment:
nth: immutable answer `return [...acc, {...token, balance}]`

Reply:
Right! maybe there is some lint for that too